flow/utils/evervigil/ever_vigil.go

```diff
@@ -0,0 +1,99 @@
+package evervigil
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	catalog "github.com/PeerDB-io/peer-flow/connectors/utils/catalog"
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgxpool"
+	"github.com/nikoksr/notify"
+	"github.com/nikoksr/notify/service/slack"
+	"github.com/sirupsen/logrus"
+)
+
+// alerting service, cool name
+type EverVigil struct {
+	notifier    *notify.Notify
+	catalogPool *pgxpool.Pool
+}
+
+type slackServiceConfig struct {
+	AuthToken  string   `json:"auth_token"`
+	ChannelIDs []string `json:"channel_ids"`
+}
+
+func NewVigil() (*EverVigil, error) {
+	catalogPool, catalogErr := catalog.GetCatalogConnectionPoolFromEnv()
+	if catalogErr != nil {
+		return nil, fmt.Errorf("error getting catalog connection pool: %w", catalogErr)
+	}
+
+	notifier := notify.New()
+
+	rows, err := catalogPool.Query(context.Background(),
+		"SELECT service_type,service_config FROM peerdb_stats.alerting_config")
+	if err != nil {
+		return nil, fmt.Errorf("failed to read everVigil config from catalog: %w", err)
+	}
+
+	var serviceType, serviceConfig string
+	_, err = pgx.ForEachRow(rows, []any{&serviceType, &serviceConfig}, func() error {
+		switch serviceType {
+		case "slack":
+			var slackServiceConfig slackServiceConfig
+			err = json.Unmarshal([]byte(serviceConfig), &slackServiceConfig)
+			if err != nil {
+				return fmt.Errorf("failed to unmarshal Slack service config: %w", err)
+			}
+
+			slackService := slack.New(slackServiceConfig.AuthToken)
+			slackService.AddReceivers(slackServiceConfig.ChannelIDs...)
+			notifier.UseServices(slackService)
+		default:
+			return fmt.Errorf("unknown service type: %s", serviceType)
+		}
+		return nil
+	})
+
+	return &EverVigil{
+		notifier:    notifier,
+		catalogPool: catalogPool,
```

Comment:
If this is only happening every 15 minutes is there a point to keeping open a connection? Pool will keep it open. Could reconnect every 15 minutes

Reply:
Yes, but that connection stays open afterwards during these 15 minute intervals between polling

Reply:
The point of a pool is to keep connections open to share between callers to avoid connection handshake overhead, some links:
1. https://github.com/jackc/pgx/issues/494
2. https://github.com/jackc/pgx/issues/205
3. `MaxConnIdleTime` defaults to 30 minutes https://github.com/jackc/pgx/blob/ba05097642a802eb5a76343f27a9a3515ed86498/pgxpool/pool.go#L134